File: services/python/edtech_service/routers/live_classes.py
from fastapi import APIRouter
from pydantic import BaseModel, Field
from datetime import datetime
from ..services.scheduling import schedule_live_class

router = APIRouter()


class ScheduleRequest(BaseModel):
    instructor_id: str = Field(..., min_length=1)
    course_id: str = Field(..., min_length=1)
    start_time: datetime


@router.post("/schedule")
async def schedule(req: ScheduleRequest):
    result = await schedule_live_class(
        instructor_id=req.instructor_id,
        course_id=req.course_id,
        start_time=req.start_time.isoformat(),
    )
    return result

File: services/node/src/routes/liveClasses.js
const express = require('express');
const { schedule_live_class } = require('../services/schedulingService');

const router = express.Router();

router.post('/schedule', async (req, res) => {
  const { instructor_id, course_id, start_time } = req.body || {};
  if (!instructor_id || !course_id || !start_time) {
    return res.status(400).json({ error: 'instructor_id, course_id, and start_time are required' });
  }
  const dt = new Date(start_time);
  if (isNaN(dt.getTime())) {
    return res.status(400).json({ error: 'start_time must be an ISO 8601 datetime string' });
  }
  try {
    const scheduled = await schedule_live_class(instructor_id, course_id, start_time);
    return res.json(scheduled);
  } catch (err) {
    req.log.error({ err }, 'Failed to schedule live class');
    return res.status(500).json({ error: 'Internal Server Error' });
  }
});

module.exports = router;
